### PR TITLE
feat: implement JAQ

### LINKED ISSUE #1321: Using JAQ as an alternative to mustache templates
**[`./jq`]** is an interesting DSL to perform JSON transformations. Embedding JQ can give developers a lot of flexibility when compared to existing mustache templates. The current mustache templates only support accessing values by path, with JQ we should be able to do much more.

[`./jq`]: https://jqlang.github.io/jq/

**Proposal**
```diff
 type Post {
    id: Int!
    userId: Int!
    title: String
-   user: User @http(path: "/users/{{value.userId}}")
+   user: User @http(path: "/users/{{.value.userId}}")
 }
```

**Technical Requirements**
- Use https://github.com/01mf02/jaq to implement JQ
- Write benchmarks comparing existing and new implementations
- Compile and store JQ templates at compile time (Blueprint generation time)

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Issue Reference(s):**  
Fixes #1321

### CODE REVIEW

File: assets/logging.png
Comment (ssddOnTop) [inline assets/logging.png:1]: idk why's this happening

File: src/json/json_like.rs [hunk 1]
```diff
@@ -186,6 +186,71 @@ impl JsonLike for async_graphql::Value {
     }
 }
 
+/*impl JsonLike for Val {
```
Comment (ssddOnTop) [inline src/json/json_like.rs:189]: need to implement or drop this

File: assets/logging.png
Comment (tusharmath) [reply assets/logging.png:1]: Check your Git settings.

Comment (webbdays) [reply assets/logging.png:1]: same issue to me.

Comment (webbdays) [reply assets/logging.png:1]: it shows changes  without any modifications actually.

**Comment by ssddOnTop** [comment]
performance is a BIG question now

**Comment by tusharmath** [comment]
@amitksingh1490 are you working on this?

**Comment by amitksingh1490** [comment]
Yes

**Comment by ssddOnTop** [comment]
I am here to help.. 

Maybe just drop comments Ill fix

File: src/json/json_like.rs [hunk 2]
```diff
@@ -186,6 +186,71 @@ impl JsonLike for async_graphql::Value {
...
+
+    fn new(value: &Self::Output) -> &Self {
+        todo!()
+    }
+
+    fn group_by<'a>(&'a self, path: &'a [String]) -> HashMap<String, Vec<&'a Self::Output>> {
+        todo!()
+    }
+}
+*/
```
Comment (tusharmath) [inline src/json/json_like.rs:253]: delete this.

**Comment by ssddOnTop** [comment]
closing in favor of #1801

**Pull request closed without merging**